src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -219,19 +238,47 @@ internal set
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
         public void RunClassInitialize(TestContext testContext)
         {
-            // If no class initialize return
-            if (this.ClassInitializeMethod == null)
+            // If no class initialize and no base class initialize, return
+            if (this.ClassInitializeMethod is null)
             {
-                return;
+                if (this.BaseClassInitializeMethodsQueue.Count == 0)
```

Comment:
use `this.BaseClassInitializeMethodsQueue.Any()` instead to make this check faster.

Reply:
Done!